lib/core/src/chain_swap.rs

```diff
@@ -367,6 +367,29 @@ impl ChainSwapHandler {
             | ChainSwapStates::TransactionLockupFailed
             | ChainSwapStates::TransactionRefunded
             | ChainSwapStates::SwapExpired => {
+                // Zero-amount Receive Chain Swaps also get to TransactionLockupFailed when user locks up funds
+                let is_zero_amount = swap.payer_amount_sat == 0;
+                if matches!(swap_state, ChainSwapStates::TransactionLockupFailed) && is_zero_amount
+                {
+                    if let Err(e) = self
+                        .swapper
+                        .get_zero_amount_chain_swap_quote(&swap.id)
+                        .map(|quote| quote.to_sat())
+                        .and_then(|quote| {
+                            info!("Got quote of {quote} sat for swap {}", &swap.id);
+
+                            self.persister.update_swap_payer_amount(&swap.id, quote)?;
+                            self.swapper
+                                .accept_zero_amount_chain_swap_quote(&swap.id, quote)
+                        })
+                    {
+                        warn!("Failed to accept the quote for swap {}: {e:?}", &swap.id);
+                    }
+
+                    // We successfully accepted the quote, the swap should continue as normal
+                    return Ok(()); // Break from TxLockupFailed branch
```

Comment:
Seems the error case arrives here too, we should allow it to be set to Refundable

Reply:
Done: 5a44cafb8be712244783f41c62b4318d1ba4937a